vcd/resource_vcd_dnat_test.go

```diff
@@ -212,16 +226,175 @@ func testAccCheckVcdDNATDestroy(s *terraform.State) error {
 }
 
 const testAccCheckVcdDnat_basic = `
+resource "vcd_network_routed" "{{.OrgVdcNetworkName}}" {
+  name         = "{{.OrgVdcNetworkName}}"
+  org          = "{{.Org}}"
+  vdc          = "{{.Vdc}}"
+  edge_gateway = "{{.EdgeGateway}}"
+  gateway      = "{{.Gateway}}"
+
+  dhcp_pool {
+    start_address = "{{.StartIpAddress}}"
+    end_address   = "{{.EndIpAddress}}"
+  }
+}
 resource "vcd_dnat" "{{.DnatName}}" {
   org          = "{{.Org}}"
   vdc          = "{{.Vdc}}"
+  network_name = "{{.OrgVdcNetworkName}}"
   edge_gateway = "{{.EdgeGateway}}"
   external_ip  = "{{.ExternalIp}}"
   port         = 7777
   internal_ip  = "10.10.102.60"
+  depends_on   = ["vcd_network_routed.{{.OrgVdcNetworkName}}"]
 }
 `
 const testAccCheckVcdDnat_tlate = `
+resource "vcd_network_routed" "{{.OrgVdcNetworkName}}" {
```

Comment:
By Terraform style, the test snippets should be all at the end of the file, after all the functions.

Reply:
Moved to bottom